src/pages/index.js

```diff
@@ -20,58 +33,205 @@ const enableValidation = (settings) => {
+        console.error(err);
+        reject(err);
+      })
+    })
   }
 );
 popUpEditProfile.setEventListeners();
 
+//PopUpWithForm
 const popUpAddItem = new PopUpWithForm("#modal-add-profile", (formData) => {
-  const name = formData.name;
-  const link = formData.link;
-  handleAddProfileFormSubmit(name, link);
+  return new Promise ((resolve, reject) => {
+    const name = formData.name;
+    const link = formData.link;
+    handleAddProfileFormSubmit(name, link)
+    .then(() => {
+      popUpAddItem.close();
+      resolve();
+  })
+  .catch((err) => {
+    console.error(err);
+    reject(err);
+   })
+  })
 });
 popUpAddItem.setEventListeners();
 
+//PopUpWithForm
+const popUpAvatar = new PopUpWithForm("#modal-avatar", (formData) => {
+  //const avatarLink = cardData.avatar;
+  const avatar = formData.avatar;
+  return new Promise ((resolve, reject) => {
+    api.updateAvatar(avatar)
+    .then((updatedAvatar) => {
+      userinfo.setAvatar(updatedAvatar.avatar);
+      popUpAvatar.close();
+      resolve();
+    })
+    .catch((err) => {
+      console.error(err);
+      reject(err);
+    })
+  })
+});
+popUpAvatar.setEventListeners();
+
+//PopUpWithConfirmation
+const popUpConfirm = new PopUpWithConfirmation("#modal-confirm-delete")
+popUpConfirm.setEventListeners();
+
+
 //Event Handlers
 function handleAddProfileFormSubmit(title, link) {
-  const newCard = createCard({ name: title, link });
-  section.addItem(newCard);
+  return new Promise((resolve, reject) => {
+  api.createCard({ name: title, link })
+  .then((card) => {
+    section.addItem(createCard(card));
+    console.log("CARD CHECK! please please work T_T.", card);
+    resolve(); 
+  })
+.catch((err) => {
+  console.error(err);
+  reject(err);
+}) 
+.finally(() => {
   popUpAddItem.close();
+})
+  });
+
 }
+
+
 function handleImageClick(data) {
   popUpWithImage.open(data);
 }
+
+function handleTrashButtonClick(item) {
+  console.log("enter trash index.js")
+  popUpConfirm.setLoading(true);
+  popUpConfirm.setSubmitCall(() => { 
+    api.deleteCard(item.getId())
+    .then(() => {
+      //const cardElement = document.querySelector(`[data-card-id="${cardId}"]`);
+      //if (cardElement){
+        popUpConfirm.close();
+        item.removeCard();
+     // }
+    //popUpConfirm.setLoading(false);
+
+    })
+    .catch((err) => {
+      console.error("Error:", err);
+     //popUpConfirm.setLoading(false);
+    })
+   .finally(() => {
+    popUpConfirm.setLoading(false);
+   })
+  })
+  popUpConfirm.close();
+  popUpConfirm.open();
+ console.log("exit trash index.js")
+}  
+
+function handleHeartButton(item) {
+  //item.isLiked = !item.isLiked;
+
+  if (item.isLiked) {
+    api.unlikeCard(item.getId())
+      .then(() => {
+        item.setLikeStatus(item.isLiked = false);
+      })
+      .catch((err) => {
+        console.error("Error:", err);
+      });
+  } else {
+    api.likeCard(item.getId())
+      .then(() => {
+        item.setLikeStatus(item.isLiked = true);
```

Comment:
You have the card in the response from the server, so you need to take `isLiked` from there